### DISCUSSION TITLE
Renovate Cannot update lockfile without public pypi

### DISCUSSION THREAD

**Discussion by frickerj** [renovatebot/renovate#22945] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate:35.129.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

My company has no access to pypi, only to our internal mirror that is inside our vpc. 
When Renovate tries to update the poetry lockfile, it tries to install virtualenv and poetry from pypi. 
It tries to install these from pypi, which causes the update to break. 
Other updates to my pyproject.toml work well, as we have set the poetry source correctly. However, the lockfiles cannot be updated. 

This is difficult/impossible to reproduce since I'm not able to create a public repo from inside our vpc. 

### Relevant debug logs

<details><summary>Logs</summary>

```
installing v2 tool poetry v1.5.0

WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 502 Bad Gateway'))': /simple/poetry/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 502 Bad Gateway'))': /simple/poetry/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 502 Bad Gateway'))': /simple/poetry/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 502 Bad Gateway'))': /simple/poetry/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 502 Bad Gateway'))': /simple/poetry/
ERROR: Could not find a version that satisfies the requirement poetry==1.5.0 (from versions: none)
ERROR: No matching distribution found for poetry==1.5.0
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment]
- https://github.com/containerbase/base/issues/1067

**Comment by rarkins** [comment]
Set binarySource=global

**Reply by frickerj** [reply]
This worked, thanks @rarkins
